<launch>

  <!-- Load model settings -->
  <arg name="load_existing_model"      default="false" />
  <arg name="model_directory"          default="$(find color_nn_segmenter)/models" />
  <arg name="model_name"               default="model" />

  <!-- Training settings -->
  <arg name="training_data_directory"  default="$(find color_nn_segmenter)/training_data" />
  <arg name="kmeans_point_per_model"   default="1" />
  <arg name="save_new_model"           default="true" />
  <arg name="save_directory"           default="$(find color_nn_segmenter)/models" />
  <!-- Ros topic settings -->
  <arg name="cloud_input_topic"        default="/camera/depth_registered/points" />
  <arg name="segmented_cloud_topic"    default="seg_label_cloud" />
  <arg name="visualized_cloud_topic"   default="visualized_cloud_topic" />

  <!-- Table segmentation settings -->
  <arg name="use_plane_segmentation"   default="true" />
  <arg name="load_table_path"          default="$(find object_on_table_segmenter)/data/table.pcd" doc="Where the table.pcd located"/>
  <arg name="table_tf"                 default="ar_marker_0" doc="Any TF frame located in the table can be used for table segmentation" />
  <arg name="update_table"             default="true" doc="Update the table data located in the data folder"/>
  <arg name="above_table_min"          default="0.01" doc="Min distance from the table"/>
  <arg name="above_table_max"          default="0.5" doc="Max distance from the table"/>
 
  <node pkg="color_nn_segmenter" type="pcl_color_segmenter" name="pcl_color_segmenter" output="screen"
    >  
    <!-- arg pass -->
    <param name="load_existing_model"       type="bool"   value="$(arg load_existing_model)" />
    <param name="model_directory"           type="str"   value="$(arg model_directory)" />
    <param name="model_name"                type="str"   value="$(arg model_name)" />
    <param name="training_data_directory"   type="str"   value="$(arg training_data_directory)" />
    <param name="kmeans_point_per_model"    type="int"   value="$(arg kmeans_point_per_model)" />
    <param name="save_new_model"            type="bool"  value="$(arg save_new_model)" />
    <param name="save_directory"            type="str"   value="$(arg save_directory)" />
    <param name="cloud_input_topic"         type="str"   value="$(arg cloud_input_topic)" />
    <param name="segmented_cloud_topic"     type="str"   value="$(arg segmented_cloud_topic)" />
    <param name="visualized_cloud_topic"    type="str"   value="$(arg visualized_cloud_topic)" />

    <param name="use_plane_segmentation"    type="bool"   value="$(arg use_plane_segmentation)" />
    <param name="table_tf"                  type="str"    value="$(arg table_tf)"/>
    <param name="update_table"              type="bool"   value="$(arg update_table)" />
    <param name="load_table_path"           type="str"    value="$(arg load_table_path)" />
    <param name="table_distance_threshold"  type="double" value="0.02"/>
    <param name="table_angular_threshold"   type="double" value="2"/>
    <param name="table_minimal_inliers"     type="double" value="7500"/>

    <param name="above_table_min"           type="double" value="$(arg above_table_min)" />
    <param name="above_table_max"           type="double" value="$(arg above_table_max)" />
  </node>

</launch>